cmd/thanos/downsample.go

```diff
@@ -51,6 +52,11 @@ func newDownsampleMetrics(reg *prometheus.Registry) *DownsampleMetrics {
 		Name: "thanos_compact_downsample_failures_total",
 		Help: "Total number of failed downsampling attempts.",
 	}, []string{"group"})
+	m.downsampleDuration = promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+		Name:    "thanos_compact_downsample_duration_seconds",
+		Help:    "Duration of downsample runs",
+		Buckets: []float64{1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192},
```

Comment:
Maybe we can start from a larger number. Please note that the time includes the block download and index verification steps so small buckets doesn't make sense here.

Reply:
Hello - that turn around was blazing fast! 

I have no reservations in making the starting intervals larger - I was merely trying to replicate the same intervals used in the `prometheus_tsdb_compaction_duration_seconds_bucket` metric currently being exposed by the compactor. 

If I'm not mistaken the `begin` variable is updated to `time.Now()` once the block has been downloaded and the index has been verified: https://github.com/thanos-io/thanos/blob/main/cmd/thanos/downsample.go#L326 

So what this should measure is the time taken to open the block and run the downsample.Downsample method.

Reply:
Yes, you are right! Sorry, the Github code review just showed the `begin := time.Now()` at the beginning so I misunderstood it.
But it would be better to double-check the buckets. Also might be good to have a label for different resolution. 5m block and 1h block are different.

Reply:
Love it - I was thinking the same. I was originally conflicted because the compaction duration histogram has no labels (like compaction level) so I thought might be better to conform to the standard. But the resolution level would result in something more meaningful.  

I'm also thinking of adding histograms for overall block upload and download times as well (in another PR). Having resolution/compaction level on that may be beneficial as well.